instruments.py

```diff
@@ -0,0 +1,239 @@
+from functools import lru_cache
+
+import numpy as np
+
+from synth import (SAMPLERATE, sine_wave, envelope_ms, release_time,
+                   lowpass_noise, bandpass_noise)
+
+
+@lru_cache()
+def silence(duration, samplerate=SAMPLERATE):
+    return np.zeros(int(duration*samplerate))
+
+
+@lru_cache()
+def default_tone(freq, duration, samplerate=SAMPLERATE):
+    # # high freq att:
+    # # 0.0 : 0.99 = 110 : 880
+    # attenuation = min(max((freq - 110) / 770 * 0.99, 0.0), 0.99)
+    # ampl = 1 - attenuation
+    ampl = 0.5
+    harmonics = [
+        # (freqmult, amplmult)
+        (1.0, 0.5),
+        # (2.0, 0.2),
+        # (4.0, 0.1),
+        (1.01, 0.3),
+        (0.2, 0.3),
+        (0.5, 0.2),
+        (0.25, 0.1),
+    ]
+    wave = sine_wave(duration, 0, 0)
+    for fm, am in harmonics:
+        wave += sine_wave(duration, freq * fm, ampl * am, samplerate)
+    atk = 15
+    dcy = 20
+    sus = 0.6
+    rel = release_time(atk, dcy, len(wave))
+    #return wave * envelope(0.1, 0.2, 0.6, 0.2, len(wave))
+    return wave * envelope_ms(atk, dcy, sus, rel, len(wave))
+
+
+@lru_cache()
+def bass(freq, duration, samplerate=SAMPLERATE):
+    ampl = 0.5
+    bass_wave = sine_wave(duration, 0, 1)
+    harmonics = [
+        (0.125, 0.5),
+        (0.25, 0.3),
+        (0.5, 0.03),
+        (1.0, 0.01)
+    ]
+    for fm, am in harmonics:
+        bass_wave += sine_wave(duration, freq * fm, ampl * am)
+
+    atk = 10
+    dcy = 0
+    sus = 1
+    rel = release_time(atk, dcy, len(bass_wave))
+    bass_wave *= envelope_ms(atk, dcy, sus, rel, len(bass_wave))
+
+    # pick_wave = sine_wave(duration, freq, ampl * 0.01)
+    # pick_wave += sine_wave(duration, freq * 2, ampl * 0.005)
+
+    # atk = 10
+    # dcy = 15
+    # sus = 0.1
+    # rel = release_time(atk, dcy, len(pick_wave))
+    # pick_wave *= envelope_ms(atk, dcy, sus, rel, len(pick_wave))
+
+    return bass_wave # + pick_wave
+
+
+@lru_cache()
+def violin(freq, duration, samplerate=SAMPLERATE):
```

Comment:
Bonus violin that accidentally sneaked in while moving stuff around.

Reply:
New sounds are always welcome!